File: Java/hayaa.rpc.service/src/main/java/hayaa/rpc/common/config/RPCConfig.java
package hayaa.rpc.common.config;

import hayaa.basemodel.model.ConfigContent;

import java.util.List;

public class RPCConfig extends ConfigContent {
    //服务注册地址
    private String serviceReg;
    //服务发布地址
    private String servicePublice;
    //心跳服务地址
    private String keepUrl;
    //服务端口
    private Integer port;
    public String getServiceReg() {
        return serviceReg;
    }

    public void setServiceReg(String serviceReg) {
        this.serviceReg = serviceReg;
    }

    public String getServicePublice() {
        return servicePublice;
    }

    public void setServicePublice(String servicePublice) {
        this.servicePublice = servicePublice;
    }

    public Integer getPort() {
        return port;
    }

    public void setPort(Integer port) {
        this.port = port;
    }

    public String getKeepUrl() {
        return keepUrl;
    }

    public void setKeepUrl(String keepUrl) {
        this.keepUrl = keepUrl;
    }

    public ProviderConfig getProviderConfiguation() {
        return providerConfiguation;
    }

    public void setProviderConfiguation(ProviderConfig providerConfiguation) {
        this.providerConfiguation = providerConfiguation;
    }

    public ConsumerConfig getConsumerConfiguation() {
        return consumerConfiguation;
    }

    public void setConsumerConfiguation(ConsumerConfig consumerConfiguation) {
        this.consumerConfiguation = consumerConfiguation;
    }
    private ProviderConfig providerConfiguation;

    private ConsumerConfig consumerConfiguation;

    /**
     * 服务提供者配置
     */
    public class ProviderConfig {
        private String name;
        //服务组
        private String group;

        public String getGroup() {
            return group;
        }
        public void setGroup(String group) {
            this.group = group;
        }
        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }
    }

    /**
     * 消费者配置
     */
    public class ConsumerConfig {
        private String name;
        private List<ServiceConfig> services;

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public List<ServiceConfig> getServices() {
            return services;
        }

        public void setServices(List<ServiceConfig> services) {
            this.services = services;
        }
    }

    /**
     * 服务配置
     */
    public class ServiceConfig {
        private String name;
        private String group;
        private String interfaceName;
        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public String getGroup() {
            return group;
        }

        public void setGroup(String group) {
            this.group = group;
        }

        public String getInterfaceName() {
            return interfaceName;
        }

        public void setInterfaceName(String interfaceName) {
            this.interfaceName = interfaceName;
        }
    }
}

File: C#/RPC/Hayaa.RPC/Config/RPCConfig.cs
using Hayaa.BaseModel;
using System;
using System.Collections.Generic;
using System.Text;

namespace Hayaa.RPC.Common.Config
{
    [Serializable]
  internal  class RPCConfig : BaseData, ConfigContent
    {
        public AppSettings AppSettings { set; get; }
        public ConnectionStrings ConnectionStrings { set; get; }
        /// <summary>
        /// 单次会话超时时间,单位毫秒
        /// </summary>
        public int SessionTimeout { set; get; }
        /// <summary>
        /// 宿主服务器的cpu核心数
        /// </summary>
        public int CpuCore { set; get; }
        //服务注册地址
        public String ServiceReg { set; get; }
        //服务发布地址
        public String ServicePublice { set; get; }
        //心跳服务地址
        public String KeepUrl { set; get; }
       
        public ProviderConfig ProviderConfiguation { set; get; }
        public ConsumerConfig ConsumerConfiguation { set; get; }
        internal class ProviderConfig
        {
            public String Name { set; get; }
            /// <summary>
            /// 服务组
            /// </summary>
            public String Group { set; get; }
            //服务发布端口
            public int Port { set; get; }

        }
        internal class ConsumerConfig
        {
            public String Name { set; get; }
            public List<ServiceConfig> Services { set; get; }
        }
        internal class ServiceConfig
        {
            public String ServerHost { set; get; }
            public int ServerPort { set; get; }
            public String Name { set; get; }
            public String Group { set; get; }
            public String InterfaceName { set; get; }
            /// <summary>
            /// 此字段仅对net有效
            /// </summary>
            public String AssemblyName { set; get; }
        }
    }
}
